common/headtracker/head_tracker.go

```diff
@@ -337,11 +371,27 @@ func (ht *HeadTracker[HTH, S, ID, BLOCK_HASH]) backfill(ctx context.Context, hea
 		fetched++
 		if ctx.Err() != nil {
 			ht.log.Debugw("context canceled, aborting backfill", "err", err, "ctx.Err", ctx.Err())
-			break
+			return fmt.Errorf("fetchAndSaveHead failed: %w", ctx.Err())
 		} else if err != nil {
 			return fmt.Errorf("fetchAndSaveHead failed: %w", err)
 		}
 	}
+
+	if head.BlockHash() != latestFinalizedHead.BlockHash() {
```

Comment:
This reaches up to finalized block. Shouldn't it be up to finalized block + history depth? Otherwise `HistoryDepth()` has no use.

Reply:
I've kept this logic as is. We actively try to backfill up to finality and keep blocks in cache up to history depth

Reply:
Certain product teams have expressed their interest in the past to have more than FinalityDepth blocks stored in Head Tracker.  I figured `HistoryDepth` could play that role, and we can actually actively fetch more blocks instead of avoiding trimming, otherwise it won't be useful to users. That's why initially I proposed changing the `HistoryDepth` name to something more representative.
But if it requires additional changes, perhaps it might get out of scope. Is there anything else we would have to change besides this part to make it work with `FinalityDepth` + `HistoryDepth`

Reply:
Changing the behaviour to backfill down to `HistoryDepth` is trivial. But It would be nice to do some testing to check how long would a fresh start take for chains like Polygon and maybe do optimisation if needed.  

To keep this PR's size manageable I'd rather introduce this feature separately.

Reply:
Not in this PR, but we could do a large backfill by batched RPC calls right?

Reply:
Yes, batching should improve performance for large block ranges, but it would be nice to measure actual improvement before introducing second way of backfilling